### ISSUE TITLE
Linode 5Gig Disk space Offer For Restarting Servers

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#3552]
If we "take a free offer" from linode we'll get a 5 gig disk space increase at the cost of a server restart and downtime. Pretty clearly a ploy for us to restart the servers for their benefit, but maybe we'll take the bribe :money_mouth_face: 

I'm thinking if we ever store files on the servers, some extra disk space will be nice. For the most part we have plenty of disk, but more certainly does not hurt.

Tracking here the servers that have had a restart and have been 'upgraded':

### Bots
- [ ] bot25_ca_usa 
- [ ] bot35_frankfurt_de
- [ ] bot45_atlanta_ga
- [ ] bot55_london_uk
- [ ] bot65_tokyo_jp

### Metrics/Monitoring
- [ ] infra-support

### Lobby and Test Lobby
- [ ] prerelease_staging
- [ ] warclub

### Forum
- [ ] TripleA-Forum

**Comment by prastle** [comment]
Nice!  @DanVanAtta 👍

**Comment by ron-murhammer** [comment]
Agree. Only 2 that are a bit tough to find good restart periods are warclub and forum but should be ok during slow times.

**Comment by prastle** [comment]
bot 35 done

**Comment by prastle** [comment]
@DanVanAtta 25 and 55 done.  I would like to say the infra works flawlessly except for 25.  Every bot server restarted themselves.  25 I had to start them again so a minor glitch there.  Good Stuff!

**Comment by DanVanAtta** [comment]
Looks like we missed a step; disks used by OS was not resized to use new space. We'll have to shut the servers down to do that. There is no hurry as we are not pressing any of the disk limits and we have the space allocated even if not usable.

**Comment by DanVanAtta** [comment]
Just the one server left, forum! Should we skip that one, or bite the bullet at some point and get the disk upgrade?

**Comment by prastle** [comment]
Y I was just going to comment on that
Hello prastle5! The following activity has recently occurred:

 * TripleAWarClub - (102160963) Resize Disk - [4611145] - Failed Thu, 19 Jul 2018 02:08:16 GMT

**Comment by prastle** [comment]
Roi can do that at any point only 1 or two users online
We have rebuilt it many times during the build its quite easy
one second
example atm 4 guest 2 users

**Comment by prastle** [comment]
But leave that for @RoiEXLab  he might want to slither in some upgrades patches at the same time ;)

**Comment by DanVanAtta** [comment]
The 'mega' server at the moment is less cost effective the mini's. I'm intending to investigate loading up more bots per server soon. For the moment we are the limit in terms of counts and memory to avoid bots coming in and out due to memory crashes.

**Comment by prastle** [comment]
@DanVanAtta  I guess I am looking at it differently so please correct me if I am wrong.  My thoughts are 2 servers for bots.  With a few spare bots on marti forum etc in case we need them on holidays.  Either way 20 should suffice generally. Thus 2 bot servers. just my 2 cents.  We can load a few extra during busy times

**Comment by DanVanAtta** [comment]
2 larger bot servers = > 20 instances @ $40 / month
7 small bot bot server => 21 instances @ $35 / month 

We need to improve those economics before large servers make sense for bots.

> We can load a few extra during busy times

We would likely still be paying for the servers, If fully de-commissioned, it's not necessarily trivial to spin up new ones. I would discourage us planning for this; we are not always likely to be there for a busy time, and IMO constantly tweaking the bots is asking for stability problems. Last, I don't think it's a particularly good use of our time to be hand-balancing the bot fleet.

**Comment by prastle** [comment]
Yes I know this... Point is allows all devs we have to focus on what maters instead of 7 little servers running around crashing all the time .... Please lanuch another mega server ill pay for it myself...

**Comment by DanVanAtta** [comment]
Are they actually crashing all the time? If so, having that problem impact more instances would be worse. 

A culture shift and theme I'd like for us to focus on going forward is to analyze crashes and to speak more specifically about them. Instead of just "crash". There are any number of reasons for crashes... It seems the last was some *weeks* ago largely due to servers being  misconfigured to have too many bot instances.

**Comment by prastle** [comment]
ok I will be blunt even tho it is the world wide web

**Comment by prastle** [comment]
Said this years ago and I will repeat (and I know nothing) You need a forced update all on same stable ...

**Comment by prastle** [comment]
Yes most are not stable or pre

**Comment by DanVanAtta** [comment]
Please create a github issue to report those events every time they occur. We need to know about the crashes and each one needs dev investigation as they can be from a different cause/reason.

**Comment by prastle** [comment]
Yes they crash all the time because of above ^^ I hope
either way you need a controlled test and managing 7 bot servers with 5 admins that have acess is a big PIA!  there is really nt much more to be said... very few devs 
you guys are a rare breed
 and mods quitting like flies

**Comment by prastle** [comment]
Simple answer is two bot servers since basically it is you me and roi that deal with them done arguing

**Comment by prastle** [comment]
@DanVanAtta

**Comment by DanVanAtta** [comment]
>  You need a forced update all on same stable ...

You say this, but there is no evidence or error message reported. Are there specific scenarios you are thinking of? It would be good to have those reported so we could discuss and solve each one at a time.

Are the other problems due to specific maps, are we messing up in our dev process and launching non-backward compatible problems? What other changes/fixes are in the latest version that are preventing so many crashes?

> and mods quitting like flies

This is off-topic.. Let's discuss in forum. Maybe look at some specific examples for what we can do to help that.

**Comment by prastle** [comment]
DEAR
it doesn't matter
I fought for years to get you more devs and mods
You have what you have
discussion for the fiftieth time makes no difference

**Comment by DanVanAtta** [comment]
> Simple answer is two bot servers since basically it is you me and roi that deal with them done arguing

I'm not necessarily arguing.. What you are asking for does not make sense as it'll be work to get that, it'll cost more, and it'll give us fewer bots. *If* also the bots are crashing, then we'll have *bigger* crashes impacting more players/games than before.

If on the other hand you'd like to report that the mini-bots are seeing crashes, it would be really interesting to look at that to judge whether it has anything to do with hardware size.

**Comment by prastle** [comment]
it is hardware size
but ask ROI
we dealt with this for 2 years

**Comment by prastle** [comment]
As a rough laymans terms and I will repeat laymans terms  you run out of mem on the little servers

**Comment by DanVanAtta** [comment]
Are we talking about problems experienced within the last 3 weeks?

Stuff from 2 years ago do not matter, the code is *so* much different now. We've fixed a number of leaks, we run the servers in an updated way, there are a number of differences.

Fixed maybe 3-5 weeks ago was a misconfiguration that was in there for about 2 months maybe which had a phantom 5th bot that was messing things up.

So it's interesting to know about problems that have been happening recently, any old problems are potentially fixed or removed.

**Comment by prastle** [comment]
last two years other than last 2 weeks

**Comment by prastle** [comment]
last two weeks after your mega server
just ocais mini bot crash

**Comment by DanVanAtta** [comment]
in fact, only the problems since the last deployment are of interest at this point.  : )

**Comment by prastle** [comment]
correct
thus kill all mini servers
and give us all a break
Ill pay foir it

**Comment by prastle** [comment]
one second

**Comment by prastle** [comment]
never mind they are in use
new server
add a new 100 ill buy

**Comment by DanVanAtta** [comment]
> and give us all a break

At some point "us" needs to be more defined. Are multiple servers really a problem? We are talking about system admin here, not community moderation. It's important I think to distinguish between those two jobs.  And seriously, papertrail is a useful tool so you can look at all servers at once.

**Comment by prastle** [comment]
one day you will call me and we will talk for an hr on the phone

**Comment by DanVanAtta** [comment]
@prastle let's get back to topic, @RoiEXLab is not going to enjoy reading through this.

Please open a ticket if you have a specific request/problem for us to look into.

I plan to check out how to get more bots running per server. Until then I'm not going to bother updating prod just to do it again soon after.

**Comment by prastle** [comment]
roi knows it all
was discussed 2 years ago :)

**Comment by prastle** [comment]
I understand your point
But my point is simple
If you add a nother mega server
the peephole that have to fix the bos need 2 ssh
not 7

**Comment by DanVanAtta** [comment]
@prastle , yes, we should likely have a TripleA call to coordinate. It is not necessarily to do so in OSS, so we need to be sure we can do so effectively. Part of that is being very specific and focusing on specific problems. Are mods leaving, we need to discuss that specifically and what we can do. Are there are crashes, we need to investigate error messages so we can see specific reasons. 

The lock-ups for the moment seem to have had nothing to do with memory but was a NPE instead. I'm actively interested to see what further issues we are going to see now we have a recent version deployed.

**Comment by prastle** [comment]
Thus way less work which I think is what you were aiming for at the beginning .. :)
@DanVanAtta

**Comment by DanVanAtta** [comment]
@prastle more fundamentally nobody should be SSH'ing to servers on a regular basis. I work in places where you log in once every 90~180 days.

**Comment by prastle** [comment]
exactly

**Comment by prastle** [comment]
BUT we are not there yet …

**Comment by DanVanAtta** [comment]
And then for multiple servers, whether it's 2 or 8, you can do a simple for  loop to send commands to all servers, with infrastructure scripts you get that. So in *this* project even, you should not be logging in to servers. If you need to run a command, you add it to infra.

**Comment by prastle** [comment]
One day you will teach me
But either way that isn't  my point... My point is all should be able to and none understand your infra :)

**Comment by prastle** [comment]
Thus
If we can make it so they all understand we have both done well!

**Comment by DanVanAtta** [comment]
>  My point is all should be able to and none understand your infra :)

I actually disagree with that. Infrastructure is realm of system administration or devops. Community moderators should not be expected to handle/understand any of how the servers operate. The fact the moderators have been logging in to servers is really them going way above and beyond the job.

On the other side, this project has even fewer system admins than it does devs.

**Comment by prastle** [comment]
Then you need simple access functions for all of them
